src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs

```diff
@@ -72,14 +73,32 @@ public static class ServiceRemotingExtensions
         public static IEnumerable<ServiceReplicaListener> CreateServiceRemotingReplicaListeners<TStatefulService>(
             this TStatefulService serviceImplementation)
             where TStatefulService : StatefulServiceBase, IService
+        {
+            return CreateServiceRemotingReplicaListeners(
+                serviceImplementation, Array.Empty<V2.Runtime.IExceptionConvertor>());
+        }
+
+        /// <summary>
+        ///  An extension method that creates an <see cref="IServiceRemotingListener"/>
+        /// for a stateful service implementation.
+        /// </summary>
+        /// <typeparam name="TStatefulService">Type constraint on the service implementation. The service implementation must
+        /// derive from <see cref="Microsoft.ServiceFabric.Services.Runtime.StatefulServiceBase"/> and implement one or more
+        /// interfaces that derive from <see cref="Microsoft.ServiceFabric.Services.Remoting.IService"/> interface.</typeparam>
+        /// <param name="serviceImplementation">A stateful service implementation.</param>
+        /// <param name="exceptionConvertors">An array of <see cref="V2.Runtime.IExceptionConvertor"/> to use in serializing and deserializing exceptions.</param>
+        /// <returns>A <see cref="IServiceRemotingListener"/> communication
+        /// listener that remotes the interfaces deriving from <see cref="Microsoft.ServiceFabric.Services.Remoting.IService"/> interface.</returns>
+        public static IEnumerable<ServiceReplicaListener> CreateServiceRemotingReplicaListeners<TStatefulService>(
+            this TStatefulService serviceImplementation, V2.Runtime.IExceptionConvertor[] exceptionConvertors)

```

Comment:
We should also have an argument for ExceptionSerializationTechnique in CreateServiceRemotingReplicaListeners/CreateServiceRemotingInstanceListeners alongside the IExceptionConvertor[] exceptionConvertors
The reason for this is that we need to keep BinaryFormatter as the default exception serialisation technique until it is deprecated. If the users do not provide any IExceptionConvertor, we then cant just set the exception serialisation technique to DCS.

If you have a better solution for this, please feel free to try that.

Reply:
Approached this in the latest commit by having two extension methods available to the user for either the stateful or stateless instance listeners. One accepts an array of IExceptionConvertor. If this is provided, the user clearly intends to use DSC, so it's configured accordingly. The other allows the user to specify either serialization technique without this IExceptionConvertor array and defaults to the BinaryFormatter (so as not to break existing consumers). Either approach resolves to an extension marked as internal so the user cannot pass arguments to it that wouldn't make sense (an array of IExceptionConvertors + use BinaryFormatting).